elide-spring/elide-spring-boot-autoconfigure/src/test/java/example/tests/HeaderIdentityTest.java

```diff
@@ -0,0 +1,246 @@
+ * Example functional test.
+ */
+@SqlMergeMode(SqlMergeMode.MergeMode.MERGE)
+@Sql(
+        executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD,
+        scripts = "classpath:db/test_init.sql",
+        statements = "INSERT INTO ArtifactGroup (name, commonName, description, deprecated) VALUES\n"
+                + "\t\t('com.example.repository','Example Repository','The code for this project', false);"
+)
+@Import(IntegrationTestSetup.class)
+@TestPropertySource(
+        properties = {
+                "elide.json-api.enableLinks=true",
+                "elide.async.export.enabled=false",
+                "elide.stripAuthorizatonHeaders=false"
+        }
+)
+@ActiveProfiles("default")
+public class HeaderIdentityTest extends IntegrationTest {
+    public static final String SORT_PARAM = "sort";
+    private String baseUrl;
+
+    @SpyBean
+    private RefreshableElide elide;
+
+    @Autowired
+    private ApplicationContext applicationContext;
+
+    @BeforeAll
+    @Override
+    public void setUp() {
+        super.setUp();
+        baseUrl = "https://elide.io/json/";
+    }
+
+    @BeforeEach
+    public void resetMocks() {
+        reset(elide.getElide());
+    }
+
+    @Test
+    public void jsonVerifyParamsAndHeadersGetTest() {
+        given()
+                .header(HttpHeaders.AUTHORIZATION, "willBeRemoved")
+                .header(HttpHeaders.PROXY_AUTHORIZATION, "willBeRemoved")
+                .header(HttpHeaders.ACCEPT_LANGUAGE, "en-US")
+                .queryParam(SORT_PARAM, "name", "description")
+                .contentType(JsonApiController.JSON_API_CONTENT_TYPE)
+                .body(
+                        datum(
+                                resource(
+                                        type("group"),
+                                        id("com.example.repository2"),
+                                        attributes(
+                                                attr("commonName", "New group.")
+                                        )
+                                )
+                        )
+                )
+                .when()
+                .post("/json/group")
+                .then()
+                .statusCode(HttpStatus.SC_CREATED);
+
+        ArgumentCaptor<MultivaluedMap<String, String>> requestParamsCaptor = ArgumentCaptor.forClass(MultivaluedMap.class);
+        ArgumentCaptor<Map<String, List<String>>> requestHeadersCleanedCaptor = ArgumentCaptor.forClass(Map.class);
+        verify(elide.getElide()).post(any(), any(), any(), requestParamsCaptor.capture(), requestHeadersCleanedCaptor.capture(), any(), any(), any());
+
+        MultivaluedHashMap<String, String> expectedRequestParams = new MultivaluedHashMap<>();
+        expectedRequestParams.put(SORT_PARAM, ImmutableList.of("name", "description"));
+        assertEquals(expectedRequestParams, requestParamsCaptor.getValue());
+
+        assertTrue(requestHeadersCleanedCaptor.getValue().containsKey("authorization"));
```

Comment:
shouldn't the cleaned headers not contain authorization and proxy-authorization? Maybe the variable is confusing..

Reply:
This is testing identity - where we disable header cleansing.

Reply:
There is another IT test where we test cleansing.

Reply:
ahh okie.. sounds good.